### PR TITLE
Sov 277/feat/withdraw fees

### PR DESCRIPTION
**Pull request by BetsyBraddock**
## Motivation and Context

PR for this ticket: https://sovryn.atlassian.net/browse/SOV-571

1. Adds ProtocolWithdrawFee entity for data analysis, with conversion to USD
2. Adds PayInterestTransfer entity for data analysis, without currency conversion because this entity is emitted frequently and it will be more efficient to do the conversion in sql
3. Adds function call signature to the Transaction entity for data analysis

## Checklist

- [ x] I have performed a self-review of my code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x ] the pull request targets the *default* branch of the repository (`development`)
- [x ] the code follows the established code style of the repository
  - x `npm run lint` passes
- [x ] the code build without any errors
  - x`npm run codegen` passes
  - x`npm run build` passes
- [ ] tests are added for the changes I made (if any source code was modified)
- [ ] documentation added or updated
- [ x] I have run the project locally and verified that there are no errors

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Fixes

fixes #[issue number], fixes #[issue number]

### CODE REVIEW

File: src/utils/Transaction.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export function createAndReturnTransaction(event: ethereum.Event): Transaction {
     transaction.from = event.transaction.from.toHexString()
     transaction.to = event.transaction.to
     transaction.value = event.transaction.value
+    transaction.functionSignature = event.transaction.input.toHexString().slice(0, 10)
```
Comment (omerzam) [inline src/utils/Transaction.ts]: why? for the swaps task?

File: src/ISovryn.ts [hunk 2]
```diff
@@ -498,3 +503,36 @@ export function handleRollover(event: RolloverEvent): void {
...
+    withdrawFees.token = token.id
+    withdrawFees.feeType = feeType
+    withdrawFees.transaction = event.transaction.hash.toHexString()
+    withdrawFees.timestamp = event.block.timestamp.toI32()
+    withdrawFees.emittedBy = event.address.toHexString()
+    withdrawFees.save()
+  }
+  const token = Token.load(event.params.token.toHexString())
+  if (token != null) {
+    createWithdrawFees(decimal.fromBigInt(event.params.tradingAmount, DEFAULT_DECIMALS), token, ProtocolFeeType.Trading, event)
```
Comment (omerzam) [inline src/ISovryn.ts]: not decimals from token entity instead of default?

File: package.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
     "dotenv": "^16.0.1",
     "eslint": "^8.21.0",
     "eslint-config-prettier": "^8.5.0",
+    "esm": "^3.2.25",
```
Comment (omerzam) [inline package.json:36]: what is this for?

File: src/utils/Transaction.ts (hunk 1 above)
Comment (BetsyBraddock) [reply src/utils/Transaction.ts]: yes, and general data analysis

File: src/ISovryn.ts (hunk 2 above)
Comment (BetsyBraddock) [reply src/ISovryn.ts]: I'll change to token decimals since token entity is already loaded

File: package.json (hunk 3 above)
Comment (BetsyBraddock) [reply package.json:36]: It's to run the partial manifest script. I uninstalled it by mistake and then re-installed as a dev dependency

File: src/utils/Transaction.ts (hunk 1 above)
Comment (omerzam) [reply src/utils/Transaction.ts]: using this is a discussion of its own, and this should be done in the relevant PR then

File: package.json (hunk 3 above)
Comment (omerzam) [reply package.json:36]: I don't see the uninstall I only see this in this PR

File: src/utils/Transaction.ts (hunk 1 above)
Comment (BetsyBraddock) [reply src/utils/Transaction.ts]: I'll remove

File: package.json [hunk 4]
```diff
@@ -41,8 +42,7 @@
     "typescript": "^4.7.4"
   },
   "dependencies": {
-    "esm": "^3.2.25",
```
Comment (BetsyBraddock) [inline package.json:44]: @omerzam the uninstall is here

**Review by omerzam — APPROVED** [review]

**Pull request merged**